overlord/servicestate/service_control_test.go

```diff
@@ -81,6 +82,26 @@ func (s *serviceControlSuite) SetUpTest(c *C) {
 
 	s.sysctlArgs = nil
 	systemctlRestorer := systemd.MockSystemctl(func(cmd ...string) (buf []byte, err error) {
+		// When calling the "snap restart" command, the initial status will be
```

Comment:
This is probably ok to have the existing unit tests running, but I think we need a targeted unit test with mocks arranges such that we execute a scenario outlined in the other comment.

Reply:
Added now